src/webui/api/torrentcreatorcontroller.cpp

```diff
@@ -0,0 +1,247 @@
+const QString KEY_PROGRESS = u"progress"_s;
+const QString KEY_SOURCE = u"source"_s;
+const QString KEY_SOURCE_PATH = u"sourcePath"_s;
+const QString KEY_STATUS = u"status"_s;
+const QString KEY_TASK_ID = u"taskID"_s;
+const QString KEY_TIME_ADDED = u"timeAdded"_s;
+const QString KEY_TIME_FINISHED = u"timeFinished"_s;
+const QString KEY_TIME_STARTED = u"timeStarted"_s;
+const QString KEY_TORRENT_FILE_PATH = u"torrentFilePath"_s;
+const QString KEY_TRACKERS = u"trackers"_s;
+const QString KEY_URL_SEEDS = u"urlSeeds"_s;
+
+namespace
+{
+    using Utils::String::parseBool;
+    using Utils::String::parseInt;
+
+#ifdef QBT_USES_LIBTORRENT2
+    BitTorrent::TorrentFormat parseTorrentFormat(const QString &str)
+    {
+        if (str == u"v1")
+            return BitTorrent::TorrentFormat::V1;
+        if (str == u"v2")
+            return BitTorrent::TorrentFormat::V2;
+        return BitTorrent::TorrentFormat::Hybrid;
+    }
+
+    QString torrentFormatToString(const BitTorrent::TorrentFormat torrentFormat)
+    {
+        switch (torrentFormat)
+        {
+        case BitTorrent::TorrentFormat::V1:
+            return u"v1"_s;
+        case BitTorrent::TorrentFormat::V2:
+            return u"v2"_s;
+        default:
+            return u"hybrid"_s;
+        }
+    }
+#endif
+
+    QString taskStatusString(const std::shared_ptr<BitTorrent::TorrentCreationTask> task)
+    {
+        if (task->isFailed())
+            return u"Failed"_s;
+
+        switch (task->state())
+        {
+        case BitTorrent::TorrentCreationTask::Queued:
+        default:
+            return u"Queued"_s;
+        case BitTorrent::TorrentCreationTask::Running:
+            return u"Running"_s;
+        case BitTorrent::TorrentCreationTask::Finished:
+            return u"Finished"_s;
+        }
+    }
+}
+
+TorrentCreatorController::TorrentCreatorController(BitTorrent::TorrentCreationManager *torrentCreationManager, IApplication *app, QObject *parent)
+    : APIController(app, parent)
+    , m_torrentCreationManager {torrentCreationManager}
+{
+}
+
+void TorrentCreatorController::addTaskAction()
+{
+    requireParams({KEY_SOURCE_PATH});
+
+    const BitTorrent::TorrentCreatorParams createTorrentParams
+    {
+        .isPrivate = parseBool(params()[KEY_PRIVATE]).value_or(false),
+#ifdef QBT_USES_LIBTORRENT2
+        .torrentFormat = parseTorrentFormat(params()[KEY_FORMAT].toLower()),
+#else
+        .isAlignmentOptimized = parseBool(params()[KEY_OPTIMIZE_ALIGNMENT]).value_or(true),
+        .paddedFileSizeLimit = parseInt(params()[KEY_PADDED_FILE_SIZE_LIMIT]).value_or(-1),
+#endif
+        .pieceSize = parseInt(params()[KEY_PIECE_SIZE]).value_or(0),
+        .sourcePath = Path(params()[KEY_SOURCE_PATH]),
+        .torrentFilePath = Path(params()[KEY_TORRENT_FILE_PATH]),
+        .comment = params()[KEY_COMMENT],
+        .source = params()[KEY_COMMENT],
+        .trackers = params()[KEY_TRACKERS].split(u'|'),
+        .urlSeeds = params()[KEY_URL_SEEDS].split(u'|')
```

Comment:
@glassez FYI it appears this new API endpoint is also using a pipe to delimit URLs. This might be worth changing before 5.0 ships.

Reply:
> @glassez FYI it appears this new API endpoint is also using a pipe to delimit URLs. This might be worth changing before 5.0 ships.

I've been thinking about this problem. Honestly, it doesn't occur to me what other character could be used instead of `|` to avoid this problem. It seems to me that this is impossible to find one. Whichever character we use, it may be inside one of the URLs of the list.
Maybe it would make sense to percent encode each URL before combining it into an array when sending, and first split the array and then percent decode each URL when receiving?